js/components/channel_messages.jsx

```diff
@@ -36,6 +36,7 @@ var ChannelMessages = React.createClass({
   },
 
   componentWillUnmount: function() {
+    MessageStore.removeListener('change', this._messageStoreChange);
```

Comment:
Ah, thats cool that React throws an error when that happens. Should be done anyway. Do you know why there is a 'two children with the same key warning'?